test/benchee/benchmark/runner_test.exs

```diff
@@ -611,97 +734,124 @@ defmodule Benchee.Benchmark.RunnerTest do
+      ])
     end
 
     test "hooks dealing with inputs still work when there is no input given" do
       me = self()
+
       %Suite{
         configuration: %{
           warmup: 1,
           time: 1,
-          before_scenario: fn(input) ->
+          before_scenario: fn input ->
             send(me, {:global_scenario, input})
             input
           end,
-          before_each: fn(input) ->
+          before_each: fn input ->
             send(me, {:global_each, input})
             input
           end,
-          after_scenario: fn(input) ->
+          after_scenario: fn input ->
             send(me, {:global_after_scenario, input})
           end
         }
       }
       |> test_suite
       |> Benchmark.benchmark("job", {fn ->
-           # still keep to make sure we only get one iteration and not too fast
-           :timer.sleep 1
-         end,
-         before_scenario: fn(input) ->
-           send(me, {:local_scenario, input})
-           input
-         end,
-         before_each: fn(input) ->
-           send(me, {:local_each, input})
-           input
-         end,
-         after_scenario: fn(input) ->
-           send(me, {:local_after_scenario, input})
-         end})
+         # still keep to make sure we only get one iteration and not too fast
+         :timer.sleep(1)
+       end,
+       before_scenario: fn input ->
+         send(me, {:local_scenario, input})
+         input
+       end,
+       before_each: fn input ->
+         send(me, {:local_each, input})
+         input
+       end,
+       after_scenario: fn input ->
+         send(me, {:local_after_scenario, input})
+       end})
       |> Benchmark.measure(TestPrinter)
 
       no_input = Benchmark.no_input()
 
-      assert_received_exactly [
-        {:global_scenario, no_input}, {:local_scenario, no_input},
-        {:global_each, no_input}, {:local_each, no_input},
-        {:global_each, no_input}, {:local_each, no_input},
-        {:local_after_scenario, no_input}, {:global_after_scenario, no_input}
-      ]
+      assert_received_exactly([
+        {:global_scenario, no_input},
+        {:local_scenario, no_input},
+        {:global_each, no_input},
+        {:local_each, no_input},
+        {:global_each, no_input},
+        {:local_each, no_input},
+        {:local_after_scenario, no_input},
+        {:global_after_scenario, no_input}
+      ])
     end
 
+    test "runs all benchmarks with all inputs exactly once as a dry run" do
+      ref = self()
+
+      inputs = %{"small" => 1, "big" => 100}
+
+      config = %{time: 0, warmup: 0, inputs: inputs, dry_run: true}
+
+      %Suite{configuration: config}
+      |> test_suite
+      |> Benchmark.benchmark("first", fn input -> send(ref, {:first, input}) end)
+      |> Benchmark.benchmark("second", fn input -> send(ref, {:second, input}) end)
+      |> Benchmark.measure(TestPrinter)
+
+      assert_received_exactly([{:first, 100}, {:first, 1}, {:second, 100}, {:second, 1}])
+    end
```

Comment:
I think that all the hooks trigger around a fry run is important enough to spec in an individual test, what do you think? :)

Reply:
Yeah, totally. I'll add that as well.